src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -767,9 +782,16 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
                 }
                 else
                 {
-                    objectWriter.SetSection(MethodCodeNode.UnixContentSection);
+                    managedCodeSection = MethodCodeNode.UnixContentSection;
+                    if (factory.Target.OperatingSystem == TargetOS.OSX)
+                    {
+                        // TODO fix: we set the section, but do not emit anything,
+                        // but the generic test fails without this code.
```

Comment:
It's not clear what this TODO tracks. Is this an issue that should be filed? What are the next steps?

`the generic test fails` might be a clear thing right now, but 6 months from now, maybe the test won't even exist anymore.